libs/common/src/state-migrations/migrations/3-fix-premium.ts

```
@@ -0,0 +1,48 @@
+// eslint-disable-next-line import/no-restricted-paths -- Used for token decoding, which are valid for days. We want the latest
+import { TokenService } from "../../auth/services/token.service";
+import { MigrationHelper } from "../migration-helper";
+import { Migrator, IRREVERSIBLE, Direction } from "../migrator";
+
+type ExpectedAccountType = {
+  profile?: { hasPremiumPersonally?: boolean };
+  tokens?: { accessToken?: string };
+};
+
+export class FixPremiumMigrator extends Migrator<2, 3> {
+  async migrate(helper: MigrationHelper): Promise<void> {
+    const accounts = await helper.getAccounts<ExpectedAccountType>();
+
+    async function fixPremium(userId: string, account: ExpectedAccountType) {
+      if (account?.profile?.hasPremiumPersonally === null && account.tokens?.accessToken != null) {
+        let decodedToken: { premium: boolean };
+        try {
+          decodedToken = await TokenService.decodeToken(account.tokens.accessToken);
+        } catch {
+          return;
+        }
```

The try/catch is new, but I'm not sure we want to silently swallow exceptions here. If there's an unexpected error, I think we want to throw loudly rather than let the user proceed in an unknown state. Or to put it another way, this would proceed as if the migration has been applied when it hasn't.

---

Yes, this was one of the few changes I made to the migrations. We definitely _do_ want to silently not perform this migration.

The result of an error here would mean that you cannot open the application, a silent failure means you're logged out. I think the recoverable situation is way better and that this was a missed bug in the original implementation.

Honestly. I'd love to set min version to 6 and not worry about any of these migrations.

---

I agree with your point about the recoverable vs. non-recoverable situations. Let's leave it then.

> Honestly. I'd love to set min version to 6 and not worry about any of these migrations.

If you wanted, you could raise the discussion more broadly about a deprecation period for this code, then drop everything outside of that period immediately. The last migration was you, 4 months ago, and the one before that was me, more than 12 months ago. A very generous period of 6 months would let us delete almost all of these migrations.

I understand if you'd rather not delay the PR further though when you've already done the work.